lib/archethic/p2p/client/connection.ex

```diff
@@ -186,20 +198,21 @@ defmodule Archethic.P2P.Client.Connection do
         :internal,
         :connect,
         :disconnected,
-        data = %{
+        _data = %{
           ip: ip,
           port: port,
           transport: transport
         }
       ) do
-    case transport.handle_connect(ip, port) do
-      {:ok, socket} ->
-        {:next_state, {:connected, socket}, data}
+    me = self()
 
-      {:error, _} ->
-        actions = [{{:timeout, :reconnect}, 500, nil}]
-        {:keep_state_and_data, actions}
-    end
+    # try to connect asynchronously so it does not block the messages coming in
+    Task.start_link(fn ->
+      # handle_connect might take a long time
+      send(me, {:connect_result, transport.handle_connect(ip, port)})
```

Comment:
Why sending a new message as the resolution of the task will already send a message with the result of function which can be catched with 
```elixir 
handle_event(:info, {ref, function_return}, :disconnected, data)
```